### ISSUE TITLE
Training on Single DRGAN results in error

### ISSUE THREAD

**Issue by harshit4084** [zhangjunh/DR-GAN-by-pytorch#10]
So when I tried to train a single DRGAN, there were some error 
1. not passing correct parameters
in DRGAN.py, line 32
        self.G = Generator(N_p=opt.N_p, N_z=opt.N_z)
result in always making multi_encoder

2.  Also in 
class Single_DRGAN(BaseModel):
there is funtion 
    def load_input(self, input):
it search of ith item of input
self.image.append(input[i]['image'])
            self.pose.append(input[i]['pose'])
            self.identity.append(input[i]['identity'])
            self.name.append(input[i]['name'])
while its just a single item.

**Comment by lorenzwurmseher** [comment]
@harshit4084 I also noticed this. I don't know if this is a question or not and you probably already fixed it on your own, but for everyone who hasn't, the fix is:
in DRGAN.py
- in line 32, add single=True
- replace lines 54-58 with:
 for i in range(len(input['pose'])):
            self.image.append(input['image'][i])
            self.pose.append(input['pose'][i])
            self.identity.append(input['identity'][i])
            self.name.append(input['name'][i])

- remove/comment lines 59-62

should work now. I just tried to remember and recap what I did to fix this problem so I might have forgotten to mention something

**Comment by harshit4084** [comment]
@lorenzwu Yes, exactly this is what I changed. It is working fine. Also, can you tell me what these parameters actually means:

        self.parser.add_argument('--N_p', type=int, default=2, help='the sum of the poses')
        self.parser.add_argument('--N_d', type=int, default=450, help='the sum of the identities')
        self.parser.add_argument('--N_z', type=int, default=50, help='the sum of the noise')

When I tried changing N_d to below 450 it always throws an error but above that like, for 500 it works fine. I didn't really get what sum means here.

**Comment by lorenzwu** [comment]
@harshit4084 ok I just checked.
- N_p is the number of poses standard is 2 because we have two poses frontal/profile 
- N_z is the dimension of the noise for the generator
- N_d is the number of identities our training set has. since we have 450 identities in the training set N_d is set to 450. This number tells the discriminator how many things it has to "classify" <self.fc = nn.Linear(320, N_d+N_p+1)> (last fc layer of the discriminator). if you lower this number to e.g. 300 you are telling the discriminator that there are 300 IDs but in reality there are still 450 IDs (lowering this number does not effect the dataloader, meaning you still have all the images from all 450 IDs) so you get an error. If you also load only the images up to ID=300 it works again (or you could simply delete the images from ID=300-450. I tested this, it worked). Setting this to a number that is higher than your actual number of IDs does not throw an error but your model should not work as intended since it thinks that there are more IDs than there actually are
